src/gui_common/thriveopedia/ThriveopediaEvolutionaryTreePage.cs

```diff
@@ -0,0 +1,158 @@
+{
+    [Export]
+    public NodePath DisabledInFreebuildPath = null!;
+
+    [Export]
+    public NodePath EvolutionaryTreePath = null!;
+
+    [Export]
+    public NodePath SpeciesDetailsLabelPath = null!;
+
+    [Export]
+    public NodePath SpeciesPreviewPath = null!;
+
+    [Export]
+    public NodePath HexPreviewPath = null!;
+
+    private readonly List<Dictionary<uint, Species>> speciesHistoryList = new();
+    private VBoxContainer disabledInFreebuild = null!;
+    private EvolutionaryTree evolutionaryTree = null!;
+    private CustomRichTextLabel speciesDetailsLabel = null!;
+    private SpeciesPreview speciesPreview = null!;
+    private CellHexesPreview hexesPreview = null!;
+
+    public override string PageName => "EvolutionaryTree";
+    public override string TranslatedPageName => TranslationServer.Translate("EVOLUTIONARY_TREE_PAGE");
+
+    public override void _Ready()
+    {
+        base._Ready();
+
+        disabledInFreebuild = GetNode<VBoxContainer>(DisabledInFreebuildPath);
+        evolutionaryTree = GetNode<EvolutionaryTree>(EvolutionaryTreePath);
+        speciesDetailsLabel = GetNode<CustomRichTextLabel>(SpeciesDetailsLabelPath);
+        speciesPreview = GetNode<SpeciesPreview>(SpeciesPreviewPath);
+        hexesPreview = GetNode<CellHexesPreview>(HexPreviewPath);
+
+        UpdateCurrentWorldDetails();
+    }
+
+    public override void UpdateCurrentWorldDetails()
+    {
+        if (CurrentGame == null)
+            return;
+
+        RebuildTree();
+    }
+
+    public override void OnNavigationPanelSizeChanged(bool collapsed)
+    {
+    }
+
+    /// <summary>
+    ///   Clear and then rebuild the evolutionary tree each time we open the page. This way, we ensure the tree is
+    ///   always up to date.
+    /// </summary>
+    public void RebuildTree()
+    {
+        if (!Visible)
+            return;
+
+        // TODO fix the tree for freebuild?
+        if (CurrentGame!.FreeBuild)
+        {
+            evolutionaryTree.Visible = false;
+            disabledInFreebuild.Visible = true;
+            return;
+        }
+
+        // Building the tree relies on the existence of a full history of generations stored in the current game. Since
+        // we only started adding these in 0.6.0, it's impossible to build a tree in older saves.
+        // TODO avoid an ugly try/catch block by actually checking the original save version?
+        try
+        {
+            evolutionaryTree.Clear();
+            speciesHistoryList.Clear();
+
+            evolutionaryTree.Init(CurrentGame.GameWorld.PlayerSpecies);
+            InitFirstGeneration();
+
+            // A possible next step would be to rebuild only when the Thriveopedia as a whole is opened.
+            foreach (var generation in CurrentGame.GameWorld.GenerationHistory)
+            {
+                var record = generation.Value;
+                evolutionaryTree.UpdateEvolutionaryTreeWithRunResults(
```

Comment:
It looks like my previous comment has something to do with this. It looks like this makes Tree believe that a species is split from another of the same generation.